flank-scripts/src/test/kotlin/flank/scripts/testartifacts/TestArtifactsUtils.kt

```diff
@@ -0,0 +1,32 @@
+package flank.scripts.testartifacts
+
+import flank.scripts.testartifacts.core.Context
+import flank.scripts.testartifacts.core.testArtifacts
+
+const val TEST_BRANCH_1 = "1-unit-test-artifacts-management"
+const val TEST_BRANCH_2 = "2-unit-test-artifacts-management"
+const val TEST_FILE_NAME = "test-file"
+const val TEST_FILE_BODY = "Hello test artifacts!!!"
+
+val testContext = Context(
+    branch = TEST_BRANCH_1
+)
+
+fun Context.prepareTestDirectory() {
+    projectRoot.testArtifacts(branch)
+        .apply { mkdir() }
+        .resolve(TEST_FILE_NAME)
+        .writeText(TEST_FILE_BODY)
+}
+
+fun Context.removeTestDirectory() {
+    projectRoot.testArtifacts(branch)
+        .deleteRecursively()
+}
+
+fun Context.removeTestArchive() {
+    projectRoot.testArtifacts.listFiles()?.forEach { file ->
+        if (file.name.startsWith(branch) && file.extension == "zip")
+            file.delete()
+    }
+}
```

Comment:
There is `org.junit.rules.TemporaryFolder` designed for exactly the same use. Have you considered it?

Reply:
I am not sure if this will be convenient for that case. The same effort probably, but If you have a clear vision how to use `org.junit.rules.TemporaryFolder` with benefits, pls do a commit or explain wider how you see implementation.